src/main/java/com/comcast/zucchini/ZucchiniRuntime.java

```diff
@@ -179,24 +183,51 @@ public void run() throws IOException {
 
             for(CucumberTagStatement statement : cf.getFeatureElements()) {
 
-                if(azt.canBarrier()) {
-                    if(azt.isParallel())
-                        order = azt.phase0.await();
-                    else
-                        order = 0;
-
-                    //reset the lock and scenario state
-                    if(order == 0) {
-                        //clear configuration here for per-scenario state
-                        azt.failedContexts.clear();
-                        azt.flexBarrier.refresh();
-                    }
+                if(statement instanceof CucumberScenarioOutline) {
+                    CucumberScenarioOutline cso = (CucumberScenarioOutline)statement;
+                    cso.formatOutlineScenario(formatter);
+                    for(CucumberExamples cucumberExamples : cso.getCucumberExamplesList()) {
+                        cucumberExamples.format(formatter);
+                        for(CucumberScenario cs : cucumberExamples.createExampleScenarios()) {
+                            if(azt.canBarrier()) {
```

Comment:
lines 192 - 204 are the same as lines 212 - 225. I think it would make sense to move to a separate method and call that method from both places.